tests/unit/util/main.c

```diff
@@ -803,4 +803,19 @@ ZTEST(util, test_CONCAT)
 	zassert_equal(CONCAT(CAT_PART1, CONCAT(CAT_PART2, CAT_PART3)), 123);
 }
 
+ZTEST(util, test_SIZEOF_FIELD)
+{
+	struct test_t {
+		uint32_t a;
+		uint8_t b;
+		uint8_t c[17];
+		int16_t d;
+	};
+
+	zassert_equal(SIZEOF_FIELD(struct test_t, a), 4);
+	zassert_equal(SIZEOF_FIELD(struct test_t, b), 1);
+	zassert_equal(SIZEOF_FIELD(struct test_t, c), 17);
+	zassert_equal(SIZEOF_FIELD(struct test_t, d), 2);
+}
+
```

Comment:
nit: these kinds of checks should really be static asserts, not runtime checks.

Reply:
Can you specify the exact assert you want me to use ?

Reply:
You can use build asserts for this presumably

e.g.

BUILD_ASSERT(SIZEOF_FIELD(struct test_t, a) == 4, "Some message about why this should be true")

The test case then becomes a build test

Reply:
Thanks, I changed the asserts.